extractor/src/main/java/org/schabi/newpipe/extractor/services/youtube/extractors/YoutubeChannelExtractor.java

```diff
@@ -64,16 +60,35 @@ public YoutubeChannelExtractor(StreamingService service, ListLinkHandler linkHan
 
     @Override
     public void onFetchPage(@Nonnull Downloader downloader) throws IOException, ExtractionException {
-        String channelUrl = super.getUrl() + CHANNEL_URL_PARAMETERS;
-        final Response response = downloader.get(channelUrl, getExtractorLocalization());
-        doc = YoutubeParsingHelper.parseAndCheckPage(channelUrl, response);
-        initialData = YoutubeParsingHelper.getInitialData(response.responseBody());
+        final String url = super.getUrl() + "/videos?pbj=1&view=0&flow=grid";
+
+        JsonArray ajaxJson;
+
+        Map<String, List<String>> headers = new HashMap<>();
+        headers.put("X-YouTube-Client-Name", Collections.singletonList("1"));
+        headers.put("X-YouTube-Client-Version",
+                Collections.singletonList(YoutubeParsingHelper.getClientVersion()));
+        final String response = getDownloader().get(url, headers, getExtractorLocalization()).responseBody();
+        if (response.length() < 50) { // ensure to have a valid response
+            throw new ParsingException("Could not parse json data for next streams");
+        }
```

Comment:
We might want to append response to the exception message. This could help us with debugging error reports later.

Apart from that, we could also add a 
`public static boolean isJsonResponseValid(String response)` to YouTubeParsingHelper.

Reply:
>We might want to append response to the exception message. This could help us with debugging error reports later.

I'll do that. The error message doesn't make sense (probably copied from somewhere), so I'll improve that as well.

>Apart from that, we could also add a
`public static boolean isJsonResponseValid(String response)` to YouTubeParsingHelper.

Is that really needed for just `response.length() < 50`?

Reply:
It is ok for now, but maybe we can validate a response in the future differently (and more accurate). We can leave it for now.

Reply:
YouTube returns `{"reload":"now"}` iirc if the client version is incorrect. I think this check isn't needed anymore now that we check whether the client version is valid and stuff in `getClientVersion()`.